src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -232,6 +239,15 @@ int ExecuteManagedAssembly(
     std::string tpaList;
     AddFilesFromDirectoryToTpaList(clrFilesAbsolutePath, tpaList);
 
+#if defined(__LINUX__)
+    void* coreclrTPLib = dlopen(coreClrTPproviderDllPath.c_str(),RTLD_NOW | RTLD_GLOBAL);
+    if (coreclrTPLib == nullptr)
```

Comment:
What was the problem you were running into that required two dlopens earlier but not anymore?